sihl/src/web_csrf.ml

```diff
@@ -82,108 +37,65 @@ let decrypt_with_salt ~salted_cipher ~salt_length =
  * HTML caching token handling
  *)
 
-let secret_to_token secret =
-  (* Randomize and scramble secret (XOR with salt) to make a token *)
-  (* Do this to mitigate BREACH attacks: http://breachattack.com/#mitigations *)
-  let secret_length = String.length secret in
-  let salt = Core_random.bytes secret_length in
-  let secret_value = secret |> String.to_seq |> List.of_seq in
-  let encrypted =
-    match xor salt secret_value with
-    | None ->
-      Logs.err (fun m -> m "Failed to encrypt CSRF secret");
-      raise @@ Crypto_failed "Failed to encrypt CSRF secret"
-    | Some enc -> enc
-  in
-  encrypted
-  |> List.append salt
-  |> List.to_seq
-  |> String.of_seq
-  (* Make the token transmittable without encoding problems *)
-  |> Base64.encode_string ~alphabet:Base64.uri_safe_alphabet
-;;
-
 let default_not_allowed_handler _ =
   Opium.Response.(of_plain_text ~status:`Forbidden "") |> Lwt.return
 ;;
 
 let middleware
     ?(not_allowed_handler = default_not_allowed_handler)
-    ?(key = "csrf")
+    ?(cookie_key = "_csrf")
     ()
   =
   let filter handler req =
-    let req, token = Web_form.consume req "csrf" in
-    let is_safe =
-      match req.Opium.Request.meth with
-      | `GET | `HEAD | `OPTIONS | `TRACE -> true
-      | _ -> false
-    in
-    let stored_secret = Web_session.find key req in
-    match token, is_safe with
-    | None, true ->
-      (* Don't check for CSRF token in safe requests *)
-      (match stored_secret with
-      | Some secret ->
-        let token = secret_to_token secret in
-        let req = set token req in
+    if Core_configuration.is_development ()
+       && Option.value
+            (Core_configuration.read_bool "NO_CSRF_BYPASS")
```

Comment:
@aronerben Should we maybe use something like `FORCE_CSRF_CHECK` to avoid the double negative?

Reply:
Yea, you're right, done.